src/python/pants/backend/docker/goals/package_image.py

```diff
@@ -377,17 +391,41 @@ def get_build_options(
     if global_build_no_cache_option:
         yield "--no-cache"
 
+    if push_image:
+        if use_buildx_option is not True:
+            raise DockerImageOptionValueError(
+                f"The `--push` flag was set on the command line, but buildx is not enabled. "
+                f"Buildx must be enabled via the Docker subsystem options in order to use this flag."
+            )
+        yield "--push"
```

Comment:
`--push` is a "shortcut" flag, we can accomplish the same thing if we set `--output=type=image,push=true`